src/shared/condition.c

```diff
@@ -578,8 +583,63 @@ static int condition_test_firmware_devicetree_compatible(const char *dtcarg) {
         return strv_contains(dtcompatlist, dtcarg);
 }
 
+static int condition_test_firmware_smbios_field(const char *expression) {
+        int r;
+        _cleanup_free_ char *field = NULL, *expected_value = NULL, *actual_value = NULL;
+        OrderOperator operator;
+
+        assert(expression);
+
+        /* Parse SMBIOS field */
+        r = extract_first_word(&expression, &field, "!<=>$", EXTRACT_RETAIN_SEPARATORS);
+        if (r < 0)
+                return r;
+        if (r == 0 || isempty(expression))
+                return -EINVAL;
+
+        /* Remove trailing spaces from SMBIOS field */
+        strstrip(field);
```

Comment:
This potentially changes the pointer, which means that it cannot be used in `_cleanup_`. You need to either define a second variable to keep the original pointer unchanged, or use a function that only removes trailing whitespace, e.g. `delete_trailing_chars()`.

Reply:
Replaced with `delete_trailing_chars()`